openml-h2o/src/main/java/com/feedzai/openml/h2o/server/H2OApp.java

```diff
@@ -326,7 +327,12 @@ private M waitForModelTrained(final Job<M> modelJob) {
      * @implNote This method needs to be synchronized. While we can't really pinpoint the exact reason, the experience
      * with tests training multiple models concurrently shows that this method is not thread safe.
      */
-    private synchronized Frame parseDataSetFile(final Path dataSetFile, final DatasetSchema schema) {
+    private synchronized Frame parseDataSetFile(final Path dataSetFile,
+                                                final DatasetSchema schema) throws ModelTrainingException{
+        if (FileUtils.sizeOf(dataSetFile.toFile()) == 0) {
+            throw new ModelTrainingException("In order to generate the model the dataset cannot be empty");
```

Comment:
Suggestion that I follow myself:  I generally always log an error before throwing an exception, motivated by the fact that the upstream code may swallow the exception (you never know what happens upstream).

Finally, please add the path of the dataSetFile to the message.